### PR TITLE
Add directoryWalk', haddocks and travis config

### PR DESCRIPTION
**Pull request by phadej**
Example build: https://travis-ci.org/phadej/machines-directory/builds/172882719

### CODE REVIEW

**Comment by phadej** [comment]
ping @aloiscochard

File: src/System/Directory/Machine.hs [hunk 1]
```diff
@@ -13,14 +13,29 @@ import System.IO.Machine
...
+--
+-- >>> runT (files <~ directoryWalk <~ source ["."])
+-- ["./.gitignore",...
 directoryWalk :: ProcessT IO FilePath FilePath
-directoryWalk = MachineT . return $ Await (\root -> f [root] []) Refl stopped where
-  f []    []      = directoryWalk
-  f xs    (y:ys)  = MachineT . return $ Yield y $ f xs ys
+directoryWalk = directoryWalk' (const True)
+
+-- | A variant of 'directoryWalk' with a predicate whether to descent
```
Comment (bitemyapp) [inline src/System/Directory/Machine.hs]: descend

File: machines-directory.cabal [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name:                machines-directory
-version:             0.2.0.9
+version:             0.2.1.0
```
Comment (bitemyapp) [inline machines-directory.cabal]: I think with these changes we go from `.1` to `1.0`, so you just need to pick your variant in this conflict.

**Comment by bitemyapp** [comment]
@phadej address comments+conflict and it's good to go.

**Comment by phadej** [comment]
@bitemyapp rebased, corrected, resolved conflicts.

**Comment by phadej** [comment]
ping @bitemyapp

**Comment by phadej** [comment]
ping @bitemyapp @aloiscochard

**Comment by bitemyapp** [comment]
@phadej I'll give this a look today

**Comment by bitemyapp** [comment]
@phadej I can't speak for @aloiscochard but I am not keen to use `cabal new-build` in the `.travis.yml`. Replace it with Stack or drop it from the PR and I'll merge it.

[codex, the app this library was written in part to support,](https://github.com/aloiscochard/codex/blob/master/.travis.yml) has a simpler `.travis.yml` you can reuse.

**Comment by phadej** [comment]
@bitemyapp, sorry, but I'm not free work force. You reviewed this one and said:

> address comments+conflict and it's good to go.

I removed `.travis.yml` all together.

**Comment by phadej** [comment]
Thanks! the release would be great,

**Comment by bitemyapp** [comment]
screen shot 2017-03-08 at 4 25 07 pm

**Comment by phadej** [comment]
Thanks again!

**Pull request merged**